packages/xo-web/src/common/select-cores-per-socket.js

```diff
@@ -0,0 +1,99 @@
+import _ from 'intl'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { injectState, provideState } from 'reaclette'
+import { omit } from 'lodash'
+
+import decorate from './apply-decorators'
+import Icon from './icon'
+import { Select } from './form'
+
+const PROP_TYPES = {
+  maxCores: PropTypes.number,
+  maxVcpus: PropTypes.number,
+  value: PropTypes.number.isRequired,
+}
+
+// https://github.com/xcp-ng/xenadmin/blob/0160cd0119fae3b871eef656c23e2b76fcc04cb5/XenModel/XenAPI-Extensions/VM.cs#L62
+const MAX_VM_SOCKETS = 16
+
+// This algorithm was inspired from: https://github.com/xcp-ng/xenadmin/blob/master/XenAdmin/Controls/ComboBoxes/CPUTopologyComboBox.cs#L116
+const SelectCoresPerSocket = decorate([
+  provideState({
+    computed: {
+      options: (state, { maxCores, maxVcpus, value }) => {
+        const options = [
+          {
+            label: _('vmChooseCoresPerSocket'),
+            value: 0,
+          },
+        ]
+
+        if (maxCores === undefined || maxVcpus === undefined) {
+          return options
+        }
+
+        const ratio = maxVcpus / MAX_VM_SOCKETS
+
+        let isSelectedValueInOptions = value === 0
+        for (
+          let coresPerSocket = maxCores;
+          coresPerSocket >= ratio;
+          coresPerSocket--
+        ) {
+          if (maxVcpus % coresPerSocket === 0) {
```

Comment:
Please add a comment explaining the rules and conditions to build the options.

Reply:
I think this [computed](https://github.com/vatesfr/xen-orchestra/pull/4187/files#diff-e1ddd042dc76cde4a34706f4b6afd508R29) help to understand what the rule.